src/data_vis.ts

```diff
@@ -103,16 +123,64 @@ class DataVisModel {
     gridViewModel: any;
     filteredTSTable: KnockoutObservableArray<any>;
 
+    constructor() {
+        this.currentPlot.subscribe(function() {
+            if (this.currentPlot() == "Impact") {
+                this.renderImpact();
+            } else if (this.currentPlot() == "Time series") {
+                this.renderTimeSeries();
+            }
+        }, this)
+        this.compare.subscribe(function() {
+            this.onUIChange(true, true);
+        }, this)
+        this.disaggregateBy.subscribe(function() {
+            this.onUIChange(true, true);
+        }, this)
+        this.cumulativePlot.subscribe(function() {
+            this.onUIChange(true, true);
+        }, this)
+        this.maxBars.subscribe(function() {
+            this.onUIChange(true, false); // this has to be false to prevent infinite regression
+        }, this)
+        this.yearFilter().selectedLow.subscribe(function() {
+            this.onUIChange(true, true);
+        }, this)
+        this.yearFilter().selectedHigh.subscribe(function() {
+            this.onUIChange(true, true);
+        }, this)
+        this.activityFilter().selectedOptions.subscribe(function() {
+            this.onUIChange(true, true);
+        }, this)
+        this.countryFilter().selectedOptions.subscribe(function() {;
+            this.onUIChange(true, true);
+        }, this)
+        this.diseaseFilter().selectedOptions.subscribe(function() {
+            this.onUIChange(true, true);
+        }, this)
+        this.vaccineFilter().selectedOptions.subscribe(function() {
+            this.onUIChange(true, true);
+        }, this)
+        this.touchstoneFilter().selectedOptions.subscribe(function() {
+            this.onUIChange(true, true);
+        }, this)
+
+        this.compareNames.subscribe(function() {
+            this.maxPlotOptions(createRangeArray(1, this.compareNames().length));
```

Comment:
I can see that this logic is a bit fiddly. As it stands the UI is sometimes redrawing the plot twice:

E.g. if you change the compare options it triggers a redraw but also triggers a change to compareNames, which triggers a change to maxBars, which triggers another redraw. 

Ideally we'd want all changes to values that affect the graph to be batched and then when no more changes are happening, the redraw is triggered. 

One way of achieving this is to create a single object containing all the relevant values, throttle changes to it using knockout's built in `rateLimit` extender, and then update the graphs when that object changes. I've implemented this here, see what you think: https://github.com/vimc/ts_data_visualisation/pull/18